src/Config/RabbitMQExtensionConfigProvider.cs

```diff
@@ -82,12 +82,25 @@ internal RabbitMQContext CreateContext(RabbitMQAttribute attribute)
                 Properties = properties,
             };
 
+            IModel channel = CreateConnectionAndChannel(hostname);

```

Comment:
Moved connection logic here because it made the most sense to me, but any feedback about this would be appreciated since I'm not quite sure how any other extension does it besides CosmosDB.

Reply:
Yes. Moving creation logic  here is right. Adding an abstraction around creating connections will allow you to create mock connections that can be used to unit tests.
Here are few pieces you need to create the abstractions:
-  IRabbitMQService, similar to [ICosmosDBService.cs](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.CosmosDB/Services/ICosmosDBService.cs)
- Add property IRabbitMQService as part of RabbitMQContext, similar to [CosmosDBContext.cs](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.CosmosDB/CosmosDBContext.cs)
- Implementation of IRabbitMQService - logic to connect to RabbitMQ needs to be move here.

Let's sync up in person. I can fill in more details